plugins/typescript/src/core/createOperationQueryFnNodes.ts

```diff
@@ -0,0 +1,150 @@
+import { camelCase } from "lodash";
+import { OperationObject } from "openapi3-ts";
+import ts, { factory as f } from "typescript";
+
+/**
+ * Create the declaration of the react-router queries.
+ *
+ * @returns Array of nodes
+ */
+export const createOperationQueryFnNodes = ({
+  operationFetcherFnName,
+  dataType,
+  errorType,
+  requestBodyType,
+  queryParamsType,
+  pathParamsType,
+  headersType,
+  variablesType,
+  fetcherFn,
+  operation,
+  url,
+  verb,
+  name,
+}: {
+  operationFetcherFnName: string;
+  dataType: ts.TypeNode;
+  errorType: ts.TypeNode;
+  requestBodyType: ts.TypeNode;
+  headersType: ts.TypeNode;
+  pathParamsType: ts.TypeNode;
+  queryParamsType: ts.TypeNode;
+  variablesType: ts.TypeNode;
+  operation: OperationObject;
+  fetcherFn: string;
+  url: string;
+  verb: string;
+  name: string;
+}) => {
+  const nodes: ts.Node[] = [];
+  if (operation.description) {
+    nodes.push(f.createJSDocComment(operation.description.trim(), []));
+  }
+
+  nodes.push(
+    f.createVariableStatement(
+      [f.createModifier(ts.SyntaxKind.ExportKeyword)],
+      f.createVariableDeclarationList(
+        [
+          f.createVariableDeclaration(
+            f.createIdentifier(camelCase(name)),
+            undefined,
+            undefined,
+            f.createArrowFunction(
+              undefined,
+              undefined,
+              variablesType.kind !== ts.SyntaxKind.VoidKeyword
+                ? [
+                    f.createParameterDeclaration(
+                      undefined,
+                      undefined,
+                      f.createIdentifier("variables"),
+                      undefined,
+                      variablesType,
+                      undefined
+                    ),
+                  ]
+                : [
+                    f.createParameterDeclaration(
+                      undefined,
+                      undefined,
+                      f.createIdentifier("signal"),
+                      f.createToken(ts.SyntaxKind.QuestionToken),
+                      f.createTypeReferenceNode(
+                        f.createIdentifier("AbortSignal")
+                      )
+                    ),
+                  ],
+              f.createTupleTypeNode([
+                f.createTypeReferenceNode(f.createIdentifier("reactQuery.QueryKey")),
+                f.createTypeReferenceNode(
+                  f.createIdentifier(
+                    "({ signal }: { signal?: AbortSignal; }) => Promise"
```

Comment:
This doesn't look like an `Identifier` 😅 #niceTry

You can use https://ts-ast-viewer.com to help you transform this arrow function into AST (so everything is typesafe and guaranteed to be valid typescript at the end)

Reply:
wow this is a nice one. i was looking for something like this but never found this one. thanks for the link.
meanwhile i added some changes for GETers and the tests ;)